src/util/string.h

```diff
@@ -634,9 +635,17 @@ std::vector<std::basic_string<T> > split(const std::basic_string<T> &s, T delim)
 
 std::wstring translate_string(std::wstring_view s, Translations *translations);
 
-std::wstring translate_string(std::wstring_view s);
+inline std::wstring translate_string(std::wstring_view s)
+{
+#if IS_CLIENT_BUILD
+	return translate_string(s, g_client_translations);
+#else
+	return translate_string(s, nullptr);
+#endif
```

Comment:
Isn't this different behaviour, so `IS_CLIENT_BUILD` unsuited?

(Looks like something where a runtime client check would make sense.)

Reply:
by the guidance I've written down: yes
in practice: no, it's totally fine

I agree it would be cleaner to undefine this helper function for server builds but unfortunately `translate_string` is used in many places and I didn't want to start a big refactor.

> (Looks like something where a runtime client check would make sense.)

A runtime check for something known at build time?
Not to mention that I can't reference `g_client_translations` on a server build, because this symbol doesn't exist.

Reply:
> > (Looks like something where a runtime client check would make sense.)
> 
> A runtime check for something known at build time? Not to mention that I can't reference `g_client_translations` on a server build, because this symbol doesn't exist.

You can define `g_client_translations = nullptr` in a server src, and remove the `#if` here. Then independent srcs will choose the translations if linked with the client object, and null if linked into server.

(Anyway, idk how this function is used, or why we have something that translates only in client builds (?). So feel free to dismiss.)